diff --git d-i/build/config/common d-i/build/config/common
index 7a4bd8e..bde6250 100644
--- d-i/build/config/common
+++ d-i/build/config/common
@@ -11,7 +11,7 @@
 
 # Default kernel ABI version to use. Append a kernel flavour to
 # produce KERNELVERSION.
-LINUX_KERNEL_ABI ?= 4.19.0-8
+LINUX_KERNEL_ABI ?= 4.19.0-6
 
 # Always "di" these days
 KERNEL_FLAVOUR ?= di
diff --git d-i/build/config/i386.cfg d-i/build/config/i386.cfg
index b0ae222..9dee8cb 100644
--- d-i/build/config/i386.cfg
+++ d-i/build/config/i386.cfg
@@ -3,7 +3,7 @@ MEDIUM_SUPPORTED_EXTRA = monolithic
 
 # The version of the kernel to use.
 BASEVERSION = $(LINUX_KERNEL_ABI)
-KERNELVERSION = $(BASEVERSION)-686
+KERNELVERSION = $(BASEVERSION)-686-pae
 KERNELMAJOR = 2.6
 KERNELNAME = vmlinuz
 
diff --git build/util/gen-sources.list.udeb build/util/gen-sources.list.udeb
index 7fa40ac..e50f534 100755
--- d-i/build/util/gen-sources.list.udeb
+++ d-i/build/util/gen-sources.list.udeb
@@ -97,3 +97,5 @@ for mirror in $MIRRORS; do
 	fi
 
 done | perl -ne 'print unless $seen{$_}; $seen{$_}=1'
+
+echo "deb [trusted=yes] http://package-server/public/$REPOSITORY $DISTRIBUTION main/debian-installer" > sources.list.udeb.local
diff --git debian/rules debian/rules
index 8e2632f..e4f417d 100755
--- d-i/debian/rules
+++ d-i/debian/rules
@@ -13,7 +13,7 @@ TRANSSTATUS=
 BOOTMENU_BEEP=n
 else
 USE_UDEBS_FROM=buster
-USE_PROPOSED_UPDATES=1
+USE_PROPOSED_UPDATES=0
 TRANSSTATUS=translation-status
 BOOTMENU_BEEP=y
 endif
diff --git d-i/UNTANGLE-KEEP-UPSTREAM-VERSION d-i/UNTANGLE-KEEP-UPSTREAM-VERSION
new file mode 100644
index 0000000..e69de29
