diff --git debian-cd/data/buster/grub-theme.in debian-cd/data/buster/grub-theme.in
index 3cb9db6f..43b54b47 100644
--- debian-cd/data/buster/grub-theme.in
+++ debian-cd/data/buster/grub-theme.in
@@ -15,7 +15,7 @@ terminal-font: "Sans Regular 12"
 #ifdef MENU0
   + hbox {
     left = 0
-    + label {text = MENU0 font = "Sans 10" color = "#c0c0c0" align = "left"}
+    + label {text = MENU0 font = "Sans 10" color = "#000000" align = "left"}
   }
 #endif
 #ifdef MENU1
@@ -54,7 +54,7 @@ terminal-font: "Sans Regular 12"
   top = 200
   height = 200
   item_font = "Sans Regular 12"
-  item_color = #c0c0c0
+  item_color = #000000
   selected_item_color = "black"
   selected_item_pixmap_style = "hl_*.png"
   item_height = 20
@@ -69,7 +69,7 @@ terminal-font: "Sans Regular 12"
     top = 0
     left = 20%
     + label {text = "Enter: " font = "Sans 10" color = "white" align = "left"}
-    + label {text = "Select             " font = "Sans 10" color = "#c0c0c0" align = "left"}
+    + label {text = "Select             " font = "Sans 10" color = "#000000" align = "left"}
 #ifdef MENU1
     + label {text = "           " font = "Sans 10" color = "white" align = "left"}
     + label {text = "Esc:   " font = "Sans 10" color = "white" align = "left"}
@@ -80,9 +80,9 @@ terminal-font: "Sans Regular 12"
     top = 0
     left = 20%
     + label {text = "E:     " font = "Sans 10" color = "white" align = "left"}
-    + label {text = "Edit Selection     " font = "Sans 10" color = "#c0c0c0" align = "left"}
+    + label {text = "Edit Selection     " font = "Sans 10" color = "#000000" align = "left"}
     + label {text = "           " font = "Sans 10" color = "white" align = "left"}
     + label {text = "C:     " font = "Sans 10" color = "white" align = "left"}
-    + label {text = "GRUB Command line" font = "Sans 10" color = "#c0c0c0" align = "left"}
+    + label {text = "GRUB Command line" font = "Sans 10" color = "#000000" align = "left"}
   }  
 }
diff --git debian-cd/tools/boot/buster/boot-x86 debian-cd/tools/boot/buster/boot-x86
index 15c5961a..b3ecd530 100644
--- debian-cd/tools/boot/buster/boot-x86
+++ debian-cd/tools/boot/buster/boot-x86
@@ -447,6 +447,11 @@ if [ -n "$KERNEL_PARAMS" ]; then
     done
 fi
 
+rm -f boot$N/isolinux/{*txt,*cat,ad*,exit*,prompt*,rq*,spk*}
+cp -f ../../../cd-root/isolinux/* boot$N/isolinux/
+rm -fr CD$N/README.*
+cp -r ../../../cd-root/* CD$N/
+
 set_default_desktop
 rm -rf cdrom
 
diff --git debian-cd/tools/boot/buster/parse_isolinux debian-cd/tools/boot/buster/parse_isolinux
--- parse_isolinux.orig	2021-05-26 13:38:10.217824041 -0400
+++ debian-cd/tools/boot/buster/parse_isolinux	2021-05-26 15:01:58.122706293 -0400
@@ -17,6 +17,9 @@
 my $tl_distro= shift or die "Need a top-level distro name (e.g. Debian)\n";
 my $tl_kernel= shift or die "Need a top-level kernel (e.g. GNU/Linux)\n";
 my $tl_version= shift or die "Need a top-level version (e.g. 7.0)\n";
+
+my $tl_title = "$tl_distro $tl_kernel $tl_version";
+my $tl_timeout = 0;
 my $theme_dir = "$outdir/boot/grub/theme";
 my @cpp_and_opts = ('cpp',
 		    '-traditional',
@@ -77,6 +80,11 @@
 	    $in_ifcpu = 1;
 	} elsif ($line =~ /default archdetect/) {
 	    $in_ifcpu = 0;
+	} elsif ($line =~ /timeout (.*)$/) {
+		$tl_timeout = $1;
+		if ( $tl_timeout > 10 ){
+			$tl_timeout = int($tl_timeout / 10);
+		}
 	} else {
 	    push(@lines, $line);
 	}
@@ -124,7 +132,7 @@
     my $filename = shift;
     my @args;
     push(@args, @cpp_and_opts);
-    push(@args, "-DTITLE=\"$tl_distro $tl_kernel $tl_version\"");
+    push(@args, "-DTITLE=\"$tl_title\"");
     for (my $i = 0; $i < $menudepth; $i++) {
 	push(@args, "-DMENU$i=\"" . $menu_title[$i] . "\"");
     }
@@ -155,18 +163,18 @@
 
 $menu{"number"} = "1";
 $menu{"label"} = "top";
-$menu{"title"} = "$tl_distro $tl_kernel UEFI Installer menu";
+$menu{"title"} = "$tl_title UEFI Installer menu";
 $menu_title[$menudepth] = $menu{"title"};
 
 my %kernel;
 my $in_kernel = 0;
 my $in_dark = 0;
 my $new_menu = 0;
+my $tl_title_colon_index = -1;
 
 $menudepth++;
 debug();
 print_indent "set theme=/boot/grub/theme/" . $menu{"number"} . "\n";
-create_theme_file($menu{"number"});
 
 foreach my $line(@lines) {
     if ($line =~ /^\s*menu begin\ (\S+)/) {
@@ -213,6 +221,16 @@
 	    $in_kernel = 0;
 	}
 	$menu{"title"} = $1;
+
+	$tl_title = $1;
+	$tl_title_colon_index = index($tl_title, ':');
+	if ( $tl_title_colon_index > -1 ){
+		$tl_title = substr($tl_title, 0, $tl_title_colon_index);
+		$tl_title =~ s/^\s+|\s+$//g;
+		$menu{"title"} = "$tl_title UEFI Installer menu";
+		$menu_title[$menudepth - 1] = $menu{"title"};
+	}
+
 	if ($new_menu) {
 	    my $hotkey = "";
 	    if ($menu{"label"} =~ m,\^(\S),) {
@@ -269,3 +287,8 @@
     print_kernel(\%kernel);
     undef %kernel;
 }
+if($tl_timeout > 0){
+	print "set timeout=$tl_timeout\n";
+	# print "set timeout $tl_timeout\n";
+}
+create_theme_file($menu{"number"});
